asciidoctorj-core/src/test/java/org/asciidoctor/jruby/ast/impl/WhenDocumentIsLoaded.java

```diff
@@ -0,0 +1,124 @@
+package org.asciidoctor.jruby.ast.impl;
+
+import org.asciidoctor.Asciidoctor;
+import org.asciidoctor.Attributes;
+import org.asciidoctor.Options;
+import org.asciidoctor.ast.Document;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class WhenDocumentIsLoaded {
+
+    private Asciidoctor asciidoctor = Asciidoctor.Factory.create();
+
+    @Test
+    public void should_return_empty_when_document_is_null() {
+        assertEmptySources(loadDocument(null));
```

Comment:
This test seems to fail for me with a NullPointerException.
It seems like this could be fixed by changing `JRubyAsciidoctor.load()` like so:
```java
    @Override
    public Document load(String content, Options options) {
        RubyHash rubyHash = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, options.map());
        return (Document) NodeConverter.createASTNode(getAsciidoctorModule().callMethod("load",
                Optional.ofNullable(content).map(rubyRuntime::newString).orElse(null), rubyHash));
    }
```

Reply:
I don't know how I run it locally, but I'd rather remove the test and deal with supporting `null` consistently in all cases in a separated PR. wdyt?

Reply:
Sure, makes sense.

Reply:
Being purist, if null was an error case, changing it now would be a breaking change.

Reply:
Since main is supposed to be the next 3.0.0 it would be allowed to introduce breaking changes :D
But of course we could keep it like this and change the test case.

Reply:
Created a story https://github.com/asciidoctor/asciidoctorj/issues/1146 and looked a but into the required changes. If it's OK, I can spend time tomorrow.